### ISSUE TITLE
Misleading error message from ShouldBeEquivalentTo when comparing null to enum value

### ISSUE THREAD

**Issue by shift-evgeny** [dennisdoomen/FluentAssertions#409]
Run the following with FluentAssertions 4.6.1:
```
new { NullableEnum = (DayOfWeek?)null }.ShouldBeEquivalentTo(new { NullableEnum = DayOfWeek.Friday });
```

Expected error message: `Expected member NullableEnum to be 5, but found <null>.`
Actual error message: `Expected member NullableEnum to be 5, but found 0.`

The actual value of the enum is being mis-reported as 0 when it's actually null.

**Comment by shift-evgeny** [comment]
Possibly related to #388, but the description of that bug is not very clear.

**Comment by dennisdoomen** [comment]
Thanks for reporting this. It seems that `ShouldBeEquivalentTo` doesn't support nullable enums at all.

**Comment by dennisdoomen** [comment]
Fixed in 4.6.2

**Comment by shift-evgeny** [comment]
Thanks, but now the message is simply "Expected 1, but found null." It no longer includes the member name, which was very useful information.

**Comment by dennisdoomen** [comment]
I noticed that as well, but kind of ignored it. 👎 to myself.

**Comment by dennisdoomen** [comment]
Fixed in 4.6.3. Because I changed to comparing nullable enums using an existing `Should` invocation, but one which doesn't support the context yet.

**Comment by shift-evgeny** [comment]
Thanks, that works! We even found some incorrect unit tests where null and 0 enum values were being treated as equal before.
